pkg/ccl/changefeedccl/sink_webhook_v2.go

```diff
@@ -209,7 +236,31 @@ func (sc *webhookSinkClient) FlushResolvedPayload(
 	})
 }
 
-// Flush implements the SinkClient interface
+// readResponseBody handles response body reading and decompression if needed.
+func (sc *webhookSinkClient) readResponseBody(res *http.Response) ([]byte, error) {
+	encoding := res.Header.Get(contentEncodingHeader)
+	if encoding == "" {
+		return io.ReadAll(res.Body)
+	}
+
+	// Convert the content-encoding header to our internal compression algorithm type.
+	algo, _, err := compressionFromString(encoding)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"webhook endpoint returned unsupported content encoding: %s", encoding)
+	}
+
+	reader, err := newDecompressionReader(algo, res.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"failed to create decompression reader for algorithm %s", algo)
+	}
+	defer reader.Close()
```

Comment:
i don't think this will close the underlying res.Body. will this leak?

Reply:
Yes, body reader needs to be closed. I've defer it. As for encoder it works with bytes.Reader and doesn't require to be closed (since it reads from memory and there is no underlying resource to free up, I guess)

Reply:
body reader issue fixed